eng/scripts/Publish-MavenPackages.ps1

```diff
@@ -121,78 +121,101 @@ foreach ($packageDetail in $packageDetails) {
     $typesOption = "-Dtypes=$($commaDelimitedTypes.Substring(1))"
   }
 
+  $shouldPublishPackage = $ShouldPublish
+  $packageReposityUrl = $RepositoryUrl
+
+  if ($packageReposityUrl -match "https://pkgs.dev.azure.com/azure-sdk/\b(internal|public)\b/*") {
+    # Azure DevOps feeds don't support staging
+    $shouldPublishPackage = $ShouldPublish -and !$StageOnly
+    $releaseType = 'AzureDevOps'
+  }
+  elseif ($packageReposityUrl -like "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+    if ($packageDetail.IsSnapshot) {
+      # Snapshots don't go to the standard maven central url
+      $packageReposityUrl = "https://oss.sonatype.org/content/repositories/snapshots/"
+      $releaseType = 'MavenCentralSnapshot'
+    }
+    elseif ($StageOnly) {
+      $releaseType = 'MavenCentralStaging'
+    }
+    else {
+      $releaseType = 'MavenCentral'
+    }
+  }
+  else {
+    throw "Repository URL must be either an Azure Artifacts feed, or a SonaType Nexus feed."
+  }
+
+  #Local GPG deployment is required when we're not going to publish a package, or when we're publishing to maven central
+  $requiresLocalGpg = !$shouldPublishPackage -or ($releaseType -eq 'MavenCentralStaging') -or ($releaseType -eq 'MavenCentral')
```

Comment:
Isn't this all cases?

Reply:
Nevermind I think I understand the logic now. For DevOps or Snapshot you will already be doing the steps that will generate the gpg files so you don't need to do it explicitly in those cases.